pkg/scalers/azure_pipelines_scaler_test.go

```diff
@@ -41,8 +41,13 @@ func TestParseAzurePipelinesMetadata(t *testing.T) {
 	for _, testData := range testAzurePipelinesMetadata {
 		t.Run(testData.testName, func(t *testing.T) {
 			var apiStub = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-				w.WriteHeader(http.StatusOK)
-				_, _ = w.Write([]byte(`{"count":1,"value":[{"id":1}]}`))
+				personalAccessToken := testData.authParams["personalAccessToken"]	
+				if personalAccessToken != "" && personalAccessToken[len(personalAccessToken)-1:] == "\n" {
```

Comment:
you are not checking the parsed PAT here, you are using the provided by the input data.
When I said about adding this here, I meant to check in the `http.Request` headers if the PAT always comes without the `/n` even the input contains it

Reply:
Made the changes, not sure how to trigger the negative check though.